### ISSUE TITLE
test fails with sympy 1.1

### ISSUE THREAD

**Issue by ericpre** [hyperspy/hyperspy#1686]
After sympy releases 1.1, some code using string substitution in sympy expression will fail. It looks to me that the reason of the failure has been discussed in sympy/sympy#12597 and we are using a pattern, which is not good.

Calling the following:
``` python
import sympy
string = 'a / b'
expr = sympy.sympify(string)
expr2 = expr.subs('a', 'x + 2')
```
with sympy 1.0 will return `('b', 'x')` for `expr2.free_symbols`.

The same in sympy 1.1 will provide different free_symbols and `expr2.free_symbols` will return `('b', 'x + 2')` instead.
The side effect of this is that this [line](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/_components/expression.py#L155) will not work as expected with hyperspy expression using substitution. The [test_expression_substitution](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/hyperspy/tests/model/test_components.py#L263) test fails with sympy 1.1.

**Comment by ericpre** [comment]
[Travis build log](https://travis-ci.org/ericpre/hyperspy/builds/246502632) showing the test failure.

**Comment by ericpre** [comment]
Fix in #1675.
